plugins/services/src/js/pages/task-details/TaskIpAddressesRow.js

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+
+import Application from "#PLUGINS/services/src/js/structs/Application";
+import ConfigurationMapLabel from "#SRC/js/components/ConfigurationMapLabel";
+import ConfigurationMapRow from "#SRC/js/components/ConfigurationMapRow";
+import ConfigurationMapValue from "#SRC/js/components/ConfigurationMapValue";
+import DCOSStore from "#SRC/js/stores/DCOSStore";
+import Pod from "#PLUGINS/services/src/js/structs/Pod";
+
+export default ({ task }) => {
+  if (task == null) {
+    return <noscript />;
+  }
+
+  let ipAddresses = [];
```

Comment:
I have seen this pattern often, is there a special reason why we don't do things like this in a  function? For me 

```js
function getIPAddressesForTask(service, task) {
  if (service instanceof Application) {
    const marathonTask = service.findTaskById(task.id) || { ipAddresses: [] };
 
    return marathonTask.ipAddresses.map(
      ipAddress => ipAddress.ipAddress
    );
  }
 
  if (service instanceof Pod) {
    const instance = service.findInstanceByTaskId(task.id);
 
    return instance ? instance.getIpAddresses() : [];
  }
  
  return [];
}
```

would be clearer, given the assumption that it can only be either an Application or a Pod. The current implementation leaves room for speculation there. For me the main advantage would be that we could export this function and test it seperately